functions/venue.js

```diff
@@ -712,6 +737,37 @@ exports.deleteVenue = functions.https.onCall(async (data, context) => {
   admin.firestore().collection("venues").doc(venueId).delete();
 });
 
+exports.voteInPoll = functions.https.onCall(
```

Comment:
@mike-lvov please assist with the right place for 
```
throw new HttpsError("has-voted", `User ${userId} has voted in ${pollId} Poll`, error);
```

Reply:
Using the FieldValue.arrayUnion method of implementation I described earlier we wouldn’t need to throw an error here. They could try and vote as many times as they wanted but it would never change the outcome.

Not sure if adding an explicit error (and thus needing to read and check the record first is worthwhile at that point.

We can just implement the ‘don’t vote twice’ logic in the frontend + use the FieldValue.arrayUnion method on the backend and it will be secure + have good UX + not need to make 2 calls on the backend to achieve it.

Reply:
Also, do we have a backend function file for chat yet? If so, that might make more sense than being in venue?

Reply:
@0xdevalias 
I can't find the separate backend function file for the chat. Looks like it's the best place for `voteInPoll` function.

Reply:
Fair. I don't think it's too hard to create a new one. But in the interest of getting this PR merged, can you just add `@debt` comments above any of these chat related functions that get added in venue saying to extract them into a new functions/chat file

`// @debt extract this into a new functions/chat backend script file`

Reply:
fixed in 0b2f14a1d4eb95897597cb8c465a2c2719fd408b